app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtFilter.java

```diff
@@ -37,31 +37,36 @@ public class TodoTxtFilter {
+
     // For any type, return a function which maps a task -> a list of string keys
-    public static GsCallback.r1<List<String>, TodoTxtParser> keyGetter(final Context context, final String type) {
+    public static GsCallback.r1<List<String>, TodoTxtTask> keyGetter(final Context context, final String type) {
         switch (type) {
             case PROJECT:
-                return TodoTxtParser::getProjects;
+                return TodoTxtTask::getProjects;
             case CONTEXT:
-                return TodoTxtParser::getContexts;
+                return TodoTxtTask::getContexts;
             case PRIORITY:
-                return task -> task.getPriority() == TodoTxtParser.PRIORITY_NONE ? Collections.emptyList() : Collections.singletonList(Character.toString(task.getPriority()));
+                return task -> task.getPriority() == TodoTxtTask.PRIORITY_NONE ? Collections.emptyList() : Collections.singletonList(Character.toString(task.getPriority()));
             case DUE:
-                final Map<TodoTxtParser.TodoDueState, String> statusMap = new HashMap<>();
-                statusMap.put(TodoTxtParser.TodoDueState.TODAY, context.getString(R.string.due_today));
-                statusMap.put(TodoTxtParser.TodoDueState.OVERDUE, context.getString(R.string.due_overdue));
-                statusMap.put(TodoTxtParser.TodoDueState.FUTURE, context.getString(R.string.due_future));
-                return task -> task.getDueStatus() == TodoTxtParser.TodoDueState.NONE ? Collections.emptyList() : Collections.singletonList(statusMap.get(task.getDueStatus()));
+                final Map<TodoTxtTask.TodoDueState, String> statusMap = new HashMap<>();
+                statusMap.put(TodoTxtTask.TodoDueState.TODAY, context.getString(R.string.due_today));
+                statusMap.put(TodoTxtTask.TodoDueState.OVERDUE, context.getString(R.string.due_overdue));
+                statusMap.put(TodoTxtTask.TodoDueState.FUTURE, context.getString(R.string.due_future));
+                return task -> task.getDueStatus() == TodoTxtTask.TodoDueState.NONE ? Collections.emptyList() : Collections.singletonList(statusMap.get(task.getDueStatus()));
         }
 
         return null;
     }
 
     // For a list of keys and a task -> key mapping, return a function which selects tasks
-    public static GsCallback.b1<TodoTxtParser> taskSelector(final Collection<String> keys, final GsCallback.r1<List<String>, TodoTxtParser> keyGetter, final boolean isAnd) {
+    public static GsCallback.b1<TodoTxtTask> taskSelector(
+            final Collection<String> keys,
+            final GsCallback.r1<List<String>, TodoTxtTask> keyGetter,
+            final boolean isAnd) {
 
-        final boolean noneIncluded = keys.remove(null);
-        final Set<String> searchSet = (keys instanceof HashSet) ? (HashSet) keys : new HashSet<>(keys);
+        final Set<String> searchSet = new HashSet<>(keys);
+        final boolean noneIncluded = searchSet.remove(null);
```

Comment:
Don't mutate the collection passed in, always use a new collection and not one which can be modified outside